src/monaco/browser/monaco-editor-provider.ts

```diff
@@ -48,13 +51,33 @@ export class MonacoEditorProvider {
                     commandService
                 }
             );
-            editor.onDispose(() => reference.dispose());
 
+            this.monacoPrefs.onPreferenceChanged(e => {
```

Comment:
It is not enough listening to changes, the editor should be initialized initially properly.